operator/go.mod

```
@@ -17,6 +17,7 @@ require (
 	github.com/spf13/pflag v1.0.9
 	github.com/stretchr/testify v1.11.1
 	go.uber.org/zap v1.27.0
+	gopkg.in/yaml.v3 v3.0.1
```

`operator/e2e/dependencies.go` brings in `gopkg.in/yaml.v3`, I'd prefer direct dependencies being SIG projects. I'm not 100% sure if the APIs are fully compatible but it should be easy enough to use the suggested dependency instead. 

Our dependencies really bloated up out of nowhere, and we should start pruning things out with time once things stabilize.

```suggestion
	sigs.k8s.io/yaml v1.6.0
```